main/src/ca/uwaterloo/flix/language/ast/TypedAst.scala

```diff
@@ -156,10 +156,14 @@ object TypedAst {
 
     case class RestrictableChoose(star: Boolean, exp: Expr, rules: List[RestrictableChooseRule], tpe: Type, eff: Type, loc: SourceLocation) extends Expr
 
+    case class ExtensibleMatch(label: Name.Label, exp1: Expr, bnd1: Binder, exp2: Expr, bnd2: Binder, exp3: Expr, tpe: Type, eff: Type, loc: SourceLocation) extends Expr
```

Comment:
swapping `label` and `exp1` makes more sense when reading it i think

Reply:
But in any case, this construct will be refactored.